### ISSUE TITLE
S3 bucket handling broken in v1.3

### ISSUE THREAD

**Issue by Nopik** [jaws-framework/JAWS#181]
I'm trying to upgrade my project to JAWS v1.3, and got some problem with S3 bucket management. `jaws new project` went fine, I added module, did `jaws dash` deploying lambda and endpoint, went fine. My lambda uses dynamo, so I added respective entries to resources-cf.json in my stage and tried `jaws deploy resources` and it fails, because bucket `myapp.com` already exists. Not in my account, though. Also, I was not asked about it during previous stages (during `jaws new project` I provided custom names for everything). So, I fixed the problem by editing default value of `aaJawsBucket` in resources-cf.json for my stage.

Therefore, I guess this can be improved somehow, this issue is to raise a flag about it.

Side question is - `Default` property of `aaJawsBucket` suggests that I can provide custom value to override default. How? Using env vars?

**Comment by Nopik** [comment]
To be clear, this seems to be regression from v1.2, where such situation had no place.

**Comment by doapp-ryanp** [comment]
@Nopik great catch.  I, like you, do not want jaws creating or managing my s3 bucket that is storing my jaws files (env vars, lambda zips, CloudFormation jsons etc).  I just haven't had time to test the non default scenario in v1.3 yet (its still a WIP btw).

@ac360 This looks like an oversight in the 1.3 jaws bucket refactor.  In addition we should not default the value for `aaJawsBucket` to the domain name.

@Nopik I'm surprised your `jaws new project` completed successfully, since the resources-cf.json creates the bucket.  Did you happen to specify the `-d` flag on `new project`?

I will make a CLI option on the `new project` command that lets you specify an already existing s3 bucket to use as the jaws bucket.  If specified, it will not be created (or managed) by resources-cf.json.  Also if specified I will set the default value of `aaJawsBucket` to the option value.  I debated removing `aaJawsBucket` all together in this scenario, but I think for consistency we need to leave it in so other items in the CF can ref it in the future if needed.

**Comment by Nopik** [comment]
I did just like the README.md says, plain old `jaws new project` and answered all the questions. No `-d` flag, so I have no idea why it didnt failed.

**Comment by doapp-ryanp** [comment]
ok thanks @Nopik . this will get fixed

**Comment by ac360** [comment]
On it right now

**Comment by ac360** [comment]
Should be fixed in 6bedcc2a1b8068e4030a215bf6897e20fef4d597

Re-open if not
